src/riak_kv_replrtq_peer.erl

```diff
@@ -0,0 +1,240 @@
+%% -------------------------------------------------------------------
+%%
+%% riak_kv_replrtq_snk: coordination of full-sync replication
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%% -------------------------------------------------------------------
+
+%% @doc Peer discovery for real-time replication
+
+-module(riak_kv_replrtq_peer).
+-ifdef(TEST).
+-include_lib("eunit/include/eunit.hrl").
+-endif.
+
+-behaviour(gen_server).
+
+-export([init/1,
+        handle_call/3,
+        handle_cast/2,
+        handle_info/2,
+        terminate/2,
+        code_change/3]).
+
+-export([start_link/0,
+            update_discovery/1,
+            prompt_discovery/1,
+            update_workers/2]).
+
+-type discovery_peer() ::
+    {riak_kv_replrtq_snk:queue_name(), [riak_kv_replrtq_snk:peer_info()]}.
+
+-define(DISCOVERY_TIMEOUT_SECONDS, 60).
+-define(AUTO_DISCOVERY_MAXIMUM_SECONDS, 900).
+-define(AUTO_DISCOVERY_MINIMUM_SECONDS, 60).
+
+-record(state, {discovery_peers = [] :: list(discovery_peer())}).
+
+%%%============================================================================
+%%% API
+%%%============================================================================
+
+-spec start_link() -> {ok, pid()}.
+start_link() ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+%% @doc
+%% Prompt for the discovery of peers
+-spec update_discovery(riak_kv_replrtq_snk:queue_name()) -> boolean().
+update_discovery(QueueName) ->
+    gen_server:call(?MODULE, {update_discovery, QueueName}, 60 * 1000).
```

Comment:
Do you really want 60 hardcoded here or rather use the macro `?DISCOVERY_TIMEOUT_SECONDS`.